### PR TITLE
Update to Pac4J v6 (#49)

### LINKED ISSUE #49: Missing pac4j v6 Support
Out-of-the-box I struggle to get spring-webflux-pac4j working with Pac4J v6

For example, there is no provided implementation of `WebContextFactory` like there is for `JEEFrameworkParameters` to expose the WebFlux `ServerWebExchange`

### PR DESCRIPTION
**Pull request by JapuDCret**
Resolves #49.

Notes
* Adapted to the new interfaces
* Added Lombok (to make it easier to use existing classes)
* Only tested locally, not fully deployed
  * Also hindered on my side, by a current bytecode incompatibility between Pac4J and the newest SAMLv5 snapshot (in `org.opensaml:opensaml-xmlsec-impl`)

### CODE REVIEW

**Comment by leleuj** [comment]
Remove the `FindBest` class and create a `org.pac4j.jee.adapter.JEEFrameworkAdapter`.

**Comment by JapuDCret** [comment]
> Remove the `FindBest` class and create a `org.pac4j.jee.adapter.JEEFrameworkAdapter`.

Done, thanks for the hint!

**Comment by leleuj** [comment]
Good work! See my comments to fix before the merge...

**Review by leleuj — CHANGES_REQUESTED** [review]

File: src/main/java/org/pac4j/framework/adapter/FrameworkAdapterImpl.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * The WebFlux framework adapter.
+ * <p>
+ * Needs to be called "FrameworkAdapterImpl",
+ * so that static initializer of {@link FrameworkAdapter} can discover it.
+ *
+ * @author Marvin Kienitz
+ * @since 3.0.0
+ */
+public class FrameworkAdapterImpl extends DefaultFrameworkAdapter {
```
[the remark this answers is not in the document]
Comment (JapuDCret) [reply src/main/java/org/pac4j/framework/adapter/FrameworkAdapterImpl.java:19]: IMHO: _eww_.
To find out, that `org.pac4j.adapter.FrameworkAdapterImpl` is the requirement for the class-name was already hard for me - but now with `org.pac4j.jee.adapter.JEEFrameworkAdapter` it's also confusing for newcomers. Maybe in the future one can implement a more flexible solution like with SPIs? Or at least have a forth valid class like `org.pac4j.jee.adapter.CustomFrameworkAdapter`.

but to get back on track: Will do!

[the remark this answers is not in the document]
Comment (leleuj) [reply src/main/java/org/pac4j/framework/adapter/FrameworkAdapterImpl.java]: You're right. This is tricky. Most people won't ever know about it.

File: src/main/java/org/pac4j/springframework/web/CallbackController.java [hunk 2]
```diff
@@ -46,18 +40,16 @@ public class CallbackController {
...
-
-            final SpringWebfluxWebContext context = (SpringWebfluxWebContext) FindBest.webContextFactory(null, config, SpringWebfluxWebContextFactory.INSTANCE).newContext(serverWebExchange);
-            final SessionStore sessionStore = FindBest.sessionStoreFactory(null, config, SpringWebfluxSessionStoreFactory.INSTANCE).newSessionStore(serverWebExchange);
+            FrameworkAdapter.INSTANCE.applyDefaultSettingsIfUndefined(config);
 
-            bestLogic.perform(context, sessionStore, config, bestAdapter, this.defaultUrl, this.renewSession, this.defaultClient);
+            config.getCallbackLogic().perform(config, this.defaultUrl, this.renewSession, this.defaultClient, frameworkParameters);
 
-            return context.getResult();
+            return Mono.empty();
```
Comment (leleuj) [inline src/main/java/org/pac4j/springframework/web/CallbackController.java:52]: This won't work.

Remove the `result` property from the `SpringWebfluxWebContext` and make the `SpringWebfluxHttpActionAdapter` return `response.writeWith(Mono.just(data))` or `Mono.empty();`

**Review by leleuj — CHANGES_REQUESTED** [review]

File: src/main/java/org/pac4j/springframework/web/CallbackController.java (hunk 2 above)
Comment (leleuj) [reply src/main/java/org/pac4j/springframework/web/CallbackController.java:52]: 3 times.

**Review by leleuj — APPROVED** [review]

**Comment by leleuj** [comment]
Excellent! Merging...

**Pull request merged**